app/src/androidTest/java/com/simplemobiletools/calculator/FakeCurrencyTask.kt

```diff
@@ -0,0 +1,28 @@
+package com.simplemobiletools.calculator
+
+import android.os.AsyncTask
+import android.util.Log
+import com.beust.klaxon.JsonObject
+import com.beust.klaxon.Parser
+import com.simplemobiletools.calculator.helpers.MoneyCalculatorImpl
+import com.simpletools.calculator.commons.helpers.Calculator
+import com.simpletools.calculator.commons.helpers.Formatter
+
+class FakeCurrencyTask(val from: String, val to: String, moneyActivity: Calculator, moneyCalculator: MoneyCalculatorImpl) : AsyncTask<Void, Void, String>() {
```

Comment:
why are we using a fake class instead of a complete mock?

Reply:
we came up with this solution as it was complex to mock the response to an http request using kotlin. To fix this we made the code in a way that would allow to mock a builder class for the web request async task. We used the spy() to override the build method to inject a faketask. this task didn’t do any request and always returned the same payload as a json string.

Reply:
Ahh. That makes sense now. Maybe you could include this in the pr description so we can quickly understand why that was done. This adds context.

Reply:
Great solution!